api/src/shared/infrastructure/i18n/i18n.js

```diff
@@ -0,0 +1,49 @@
+import path from 'node:path';
+import * as url from 'node:url';
+
+import { I18n } from 'i18n';
+
+import { SUPPORTED_LOCALES } from '../../domain/constants.js';
+
+const __dirname = url.fileURLToPath(new URL('.', import.meta.url));
+const translationsFolder = path.resolve(path.join(__dirname, '../../../../translations'));
+
+const supportedLocales = SUPPORTED_LOCALES.map((supportedLocale) => supportedLocale.toLowerCase());
+
+export const options = {
+  locales: ['en', 'fr', 'es', 'nl'],
+  directory: translationsFolder,
+  defaultLocale: 'fr',
+  queryParameter: 'lang',
+  languageHeaderField: 'Accept-Language',
+  objectNotation: true,
+  updateFiles: false,
+  mustacheConfig: {
+    tags: ['{', '}'],
+    disable: false,
+  },
+};
+
+// This is an optimization to avoid settings a new instance each time
+// we need to use i18n.
+const i18nInstances = {};
+
+export function getI18n() {
+  if (!i18nInstances.global) {
+    i18nInstances.global = new I18n(options);
+  }
+  return i18nInstances.global;
+}
+
+export function getI18nForLocale(locale) {
+  if (!locale || !supportedLocales.includes(locale?.toLowerCase())) {
```

Comment:
**question** : est ce la lib `i18n` va prendre la locale `nl` par défaut si on lui passe `nl-BE` ou `fr` dans le cas de `fr-BE`?